packages/next-auth/src/core/lib/cookie.ts

```diff
@@ -120,7 +120,7 @@ export class SessionStore {
     option: CookieOption,
     req: {
       cookies?: Record<string, string>
-      headers?: Record<string, string> | IncomingHttpHeaders
+      headers?: Headers | IncomingHttpHeaders | Record<string, string>
```

Comment:
Doesn't `Headers` require node 18?

Reply:
Interesting. `Headers` and other Fetch API interfaces are also available on Node 16 with the `--experimental-fetch` flag: https://nodejs.org/en/blog/release/v16.15.0/

Reply:
I had some testing in jest of some API helpers which started breaking with the new patch. Fixed it by adding `import 'next';` in the top.

Reply:
This one is old, but came up recently so. Next.js polyfills `Headers` for you, if it's missing:

https://github.com/vercel/next.js/blob/canary/packages/next/server/node-polyfill-fetch.js